src/Symfony/Component/Notifier/Bridge/Iqsms/IqsmsTransportFactory.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Iqsms;
+
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Oleksandr Barabolia <alexandrbarabolya@gmail.com>
+ *
+ * @experimental in 5.3
+ */
+final class IqsmsTransportFactory extends AbstractTransportFactory
+{
+    /**
+     * @return IqsmsTransport
+     */
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+        $login = $this->getUser($dsn);
+        $password = $this->getPassword($dsn);
+        $from = $dsn->getOption('from');
+
+        if (!$from) {
+            throw new IncompleteDsnException('Missing from.', $dsn->getOriginalDsn());
+        }
+
+        $host = 'default' === $dsn->getHost() ? null : $dsn->getHost();
+        $port = $dsn->getPort();
+
+        if ('iqsms' === $scheme) {
+            return (new IqsmsTransport($login, $password, $from, $this->client, $this->dispatcher))
+                ->setHost($host)
+                ->setPort($port);
+        }
+
+        throw new UnsupportedSchemeException($dsn, 'iqsms', $this->getSupportedSchemes());
+    }
```

Comment:
Please pull up the scheme check like here, to first check if the scheme is supported and afterwards for missing options. Thanks

https://github.com/symfony/symfony/blob/7d846d3e82f4e3dabec7d4b5300f50a1dd9ed7d6/src/Symfony/Component/Notifier/Bridge/Sinch/SinchTransportFactory.php#L27-L47